pkg/mq/msgstream/mq_msgstream.go

```diff
@@ -488,6 +488,25 @@ func (ms *mqMsgStream) Seek(ctx context.Context, msgPositions []*msgpb.MsgPositi
 	return nil
 }
 
+func (ms *mqMsgStream) ResetProducer() {
+	log.Info("start to reset producer", zap.Int("producer num", len(ms.producers)))
+	ms.streamCancel()
+	ms.closeRWMutex.Lock()
+	defer ms.closeRWMutex.Unlock()
+	if !atomic.CompareAndSwapInt32(&ms.closed, 0, 1) {
```

Comment:
The flag needs to be set back to 0 after you reopen the stream, but the context canceled is not invertible, make sure all internal states works well after you reopen the stream